disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -35,6 +39,13 @@ maybeGen g = sized $ \s ->
     (1, return Nothing),
     (s, Just <$> resize (s `div` 2) g)]
 
+-- | Wait for a generated `Just` value
+--
+-- Use _only_ in case of emergencies when you have no other way to get an `a` safely
+genFromMaybe :: Gen (Maybe a) -> Gen a
```

Comment:
@domdere Doesn't compile, was it your version of [suchThat](http://hackage.haskell.org/package/QuickCheck-2.8.2/docs/src/Test-QuickCheck-Gen.html#suchThat)?
Anyway, personally I wouldn't add this combinator to disorder: if the original `Gen (Maybe a)` generates `Just`'s only just do `fromJust <$>` but if "Nothing" is possible, the generator could always fail depending on the `size` so the test becomes unreliable. The best option I guess is to lift entire test code into `Maybe` and check it at the end of the test.